python/util/embed_resource.py

```diff
@@ -0,0 +1,463 @@
+from typing import Any, Dict  # noqa # pylint: disable=unused-import
+
+
+def dict_append(embed, new_dict):
+    """ Extend an existing embed dictionary with new keys and key values """
+    embed_copy = embed.copy()
+    for key, value in new_dict.items():
+        embed_copy[key] = value
+    return embed_copy
```

Comment:
why not just the usual`{**x, **y}` idiom?

Reply:
the docstring is wrong anyway, you are not extending an existing dict here, you
are building a new one

Reply:
This function was copied over from the previous code could be fixed in the future.

Reply:
can you give me an example of this syntax?

Reply:
uuhh, how do you mean, it's literally just the one i typed above? replace x and y with any other variable names you like.

Reply:
so something like:

def dict_append(embed, new_embed):
    """ Returns a new dict with keys and values from both embed and new_embed """
    return {**embed, **new_embed}

Reply:
yeah, but it's shorter to type `{**,**}` than it is to type `dict_append()` so people wouldn't typically bother defining such a function to start with.

there might be some subtle difference in behaviour between the two forms that i'm not seeing?